File: backend/hr/human_resources/models/absence.py
from django.db import models


class Absence(models.Model):
    id = models.AutoField(primary_key=True)
    requester = models.ForeignKey(
        'human_resources.userprofile', related_name='absences', on_delete=models.CASCADE)
    state = models.CharField(max_length=60)
    kind = models.CharField(max_length=60)
    start = models.CharField(max_length=60)
    ending = models.CharField(max_length=60)

    def __str__(self):
        return self.id

File: frontend/hr-ui/src/app/core/models/absence/absence.ts
export class Absence {
    public id?: number
    public requester?: number
    public state?: string
    public kind?: string
    public start?: string
    public ending?: string

    public constructor(requester: number, state: string, kind: string, start: string, ending: string){
        this.requester = requester;
        this.state = state;
        this.kind = kind;
        this.start = start;
        this.ending = ending;
    }
}